From bb090d3c930a217c5d6e7b0575076c54ff3ee770 Mon Sep 17 00:00:00 2001
From: DaveTBlake <oak99sky@yahoo.co.uk>
Date: Fri, 22 Feb 2019 13:03:58 +0000
Subject: [PATCH] Stop playing current file when playlist player has stopped.
 Playlistplayer sends a GUI_MSG_PLAYLISTPLAYER_STOPPED message when aborting
 playback when next from last item in list, or repeat one and song is invalid.
 This ensures that the player stops too.

---
 xbmc/Application.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index a8ff2ceb10bf..1a6b8fa3aa3b 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -3896,6 +3896,8 @@ bool CApplication::OnMessage(CGUIMessage& message)
   case GUI_MSG_PLAYLISTPLAYER_STOPPED:
     m_itemCurrentFile->Reset();
     CServiceBroker::GetGUI()->GetInfoManager().ResetCurrentItem();
+    if (m_appPlayer.IsPlaying())
+      StopPlaying();
     PlaybackCleanup();
     return true;
 
